### PR TITLE
Add Search and Rescue 2

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* https://github.com/flatpak/xdg-desktop-portal/issues/536 still missing
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/SearchAndRescue2/sar2/issues/11
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/SearchAndRescue2/sar2/pull/36

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build io.github.searchandrescue2.sar2

**Comment by Mailaender** [comment]
bot, build io.github.searchandrescue2.sar2

File: io.github.searchandrescue2.sar2.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  - name: sar2
+    buildsystem: simple
+    build-commands:
+      - scons --prefix=${FLATPAK_DEST} --ldflags=-L${FLATPAK_DEST}/lib --optflags=-I${FLATPAK_DEST}/include
+    post-install:
+      - mkdir -p ${FLATPAK_DEST}/share/sar2/
+      - cp -ar data/* ${FLATPAK_DEST}/share/sar2/
+      - mkdir -p ${FLATPAK_DEST}/bin
+      - cp -a bin/sar2 ${FLATPAK_DEST}/bin
```
Comment (hfiguiere) [inline io.github.searchandrescue2.sar2.yaml:63]:
```suggestion
      - install -Dm755 bin/sar2 -t ${FLATPAK_DEST}/bin
```

**Comment by Mailaender** [comment]
bot, build io.github.searchandrescue2.sar2

**Comment by barthalion** [comment]
It doesn't work on Wayland.

**Comment by Mailaender** [comment]
bot, build io.github.searchandrescue2.sar2

**Comment by barthalion** [comment]
```
 Traceback (most recent call last):
Cloning https://github.com/Mailaender/flathub.git (branch: sar2)
  File "/entrypoint.py", line 247, in <module>
    main()
  File "/entrypoint.py", line 171, in main
    clone.update_submodules(init=True)
  File "/usr/local/lib64/python3.11/site-packages/pygit2/repository.py", line 199, in update_submodules
    payload.check_error(err)
  File "/usr/local/lib64/python3.11/site-packages/pygit2/callbacks.py", line 98, in check_error
    check_error(error_code)
  File "/usr/local/lib64/python3.11/site-packages/pygit2/errors.py", line 65, in check_error
    raise GitError(message)
_pygit2.GitError: authentication required but no callback set
```

Have you done something to your repo? For whatever reason, the merge fails that way

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "shared-modules"]
+	path = shared-modules
+	url = git@github.com:flathub/shared-modules.git
```
Comment (hfiguiere) [inline .gitmodules:3]: this MUST be a https URL

**Comment by barthalion** [comment]
I guess that's what I get for using a relatively low-level library. We override SSH-based GitHub URLs on runners.

**Comment by hfiguiere** [comment]
this is a check that could be done.

File: .gitmodules (hunk 2 above)
Comment (Mailaender) [reply .gitmodules]: Sorry

**Comment by Mailaender** [comment]
bot, build io.github.searchandrescue2.sar2

**Pull request closed without merging**